src/metal/window-proxy.ts

```diff
@@ -43,19 +47,35 @@ let W: WindowProxy = {
   getScrollLeft: nop,
   getHeight: nop,
   getWidth: nop,
+  rootsList: [],
   onWindowIsDirtyListeners: [],
   rAF: hasRAF ? window.requestAnimationFrame.bind(window) : (callback: Function) => { callback(); },
+  bindRootScrollEvent: (root: Element) => { root.addEventListener('scroll', () => W.callIsDirtyListeners(), false); },
+  callIsDirtyListeners() {
+    if (this.onWindowIsDirtyListeners.length <= 0) { return; }
+
+    this.onWindowIsDirtyListeners.forEach((obj: any) => {
+      let { fn, scope } = obj;
+      fn.call(scope);
+    });
+  },
   meta: {
     width: 0,
     height: 0,
     scrollTop: 0,
     scrollLeft: 0
   },
-  __destroy__() {
+  disconnectIsDirtyListener(id: string) {
+    this.onWindowIsDirtyListeners = this.onWindowIsDirtyListeners.filter((obj: any) => {
+      return obj.id !== id;
+    });
+  },
+  disconnectAll() {
```

Comment:
This is somewhat misleading, as it does not actually disconnect, merely orphans the above attached event listeners.

This orphaning of added (but not removed) scroll event handlers on custom roots may itself problematic.